### PR TITLE
style(databases-upload-form): update Upload Form cosmetics  

### LINKED ISSUE #30737: CSV, Columnar, and Excel Upload Form cosmetic updates
## Screenshot

Screenshot 2024-10-16 at 4 04 20 PM
Screenshot 2024-10-16 at 4 07 02 PM

## Description

The following updates need to be made in the CSV, columnar, and Excel upload forms: 
* Move the "Preview uploaded file" toggle to be left aligned with the rest of the form fields, under the file upload option and above the Columns field 
* In the first item of "General information" field, it should read "CSV file" and "Columnar file" (file is not capitalized, CSV is, Columnar is) 
* All headers and form fields should be sentence case and not title case. For example, it should be: 
   * CSV upload rather than CSV Upload
   * Columnar upload rather than Columnar Upload
   * Table name rather than Table Name
   * File settings
   * If table already exists
   * Columns to be parsed as dates
   * Decimal character
   * Columns to read
   * Column data types
   * Header row
   * Rows to read
   * Skip rows
   * Data imported rather than Data Imported in the success toast 

To access these forms, connect a database and in the database settings, enable the "Allow file uploads to database" option in the Advanced > Security section. Then you should see an "Upload file to database" option at the top right of the Databases page.

### PR DESCRIPTION
**Pull request by vine-trellis**
### SUMMARY

This PR updates the cosmetics of the "Upload file to database" form:

- Labels are sentence cased
- Preview uploaded file switch left-aligned and moved between file upload and Columns field

Fixes #30737 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:
image
image

After:
image
image

### TESTING INSTRUCTIONS

1. Log into dashboard
2. Navigate to Settings/Database Connections
3. Connect to a database and enable `Allow file uploads to database`
4. Click `Upload file to database` dropdown and select an option to open the form modal
5. Verify label casing and preview switch position

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #30737
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/databases/UploadDataModel/index.tsx [hunk 1]
```diff
@@ -173,6 +173,12 @@ const allowedExtensionsToAccept = {
   columnar: '.parquet, .zip',
 };
 
+const extensionsToLabel: Record<UploadType, string> = {
+  csv: 'CSV',
```
Comment (mistercrunch) [inline superset-frontend/src/features/databases/UploadDataModel/index.tsx:177]: did a double take on whether this needs to be wrapped with a18n, but i think it's right to NOT make these labels translatable

**Review by mistercrunch — APPROVED** [review]
LGTM - thanks for the contribution!

**Comment by geido** [comment]
Thanks for the PR @vine-trellis! Would you please add some tests cases for these labels?

**Review by rusackas — APPROVED** [review]
Thanks!

**Comment by rusackas** [comment]
Oops, merged before the response about test cases... but I think this is something we should handle more broadly via a lint rule, no?

**Comment by geido** [comment]
@vine-trellis feel free to bring in more test cases or add a lint rule as Evan suggested. That would be super helpful!

**Pull request merged**